gnd/src/main/java/com/google/android/gnd/ui/map/gms/BasemapSelectorMapAdapter.java

```diff
@@ -0,0 +1,212 @@
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.UiSettings;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gnd.model.feature.Point;
+import com.google.android.gnd.ui.map.Extent;
+import com.google.android.gnd.ui.map.ExtentSelector;
+import com.google.common.collect.ImmutableSet;
+import com.google.maps.android.data.Feature;
+import com.google.maps.android.data.geojson.GeoJsonFeature;
+import com.google.maps.android.data.geojson.GeoJsonLayer;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+
+import io.reactivex.Observable;
+import io.reactivex.subjects.BehaviorSubject;
+import io.reactivex.subjects.PublishSubject;
+
+import static com.google.android.gms.maps.GoogleMap.OnCameraMoveStartedListener.REASON_DEVELOPER_ANIMATION;
+import static java8.util.stream.StreamSupport.stream;
+
+public class BasemapSelectorMapAdapter implements ExtentSelector {
+
+  private static final String TAG = GoogleMapsMapAdapter.class.getSimpleName();
+  private static final String GEO_JSON_FILE = "gnd-geojson.geojson";
+  private final GoogleMap map;
+  private final Context context;
+
+  private final PublishSubject<Point> dragInteractionSubject = PublishSubject.create();
+  private final BehaviorSubject<Point> cameraPositionSubject = BehaviorSubject.create();
+  private final PublishSubject<Extent> extentsSubject = PublishSubject.create();
+
+  @Nullable private LatLng cameraTargetBeforeDrag;
+  private GeoJsonLayer extentSelectionLayer;
+  private HashMap<String, Extent> availableExtents;
+
+  public BasemapSelectorMapAdapter(GoogleMap map, Context context) {
+    this.map = map;
+    this.context = context;
+    map.setMapType(GoogleMap.MAP_TYPE_HYBRID);
+    UiSettings uiSettings = map.getUiSettings();
+    uiSettings.setRotateGesturesEnabled(false);
+    uiSettings.setTiltGesturesEnabled(false);
+    uiSettings.setMyLocationButtonEnabled(false);
+    uiSettings.setMapToolbarEnabled(false);
+    uiSettings.setCompassEnabled(false);
+    uiSettings.setIndoorLevelPickerEnabled(false);
+    map.setOnCameraIdleListener(this::onCameraIdle);
+    map.setOnCameraMoveStartedListener(this::onCameraMoveStarted);
+    map.setOnCameraMoveListener(this::onCameraMove);
+    onCameraMove();
+  }
+
+  @Override
+  public void renderExtentSelectionLayer() {
+    File file = new File(context.getFilesDir(), GEO_JSON_FILE);
+
+    try {
+      InputStream is = new FileInputStream(file);
+      BufferedReader buf = new BufferedReader(new InputStreamReader(is));
+      String line = buf.readLine();
+      StringBuilder sb = new StringBuilder();
+      while (line != null) {
+        sb.append(line).append("\n");
+        line = buf.readLine();
+      }
```

Comment:
Wasn't there a cleaner way to load the file? Something related to JSONTokenizer?

Reply:
There's definitely a better way--I couldn't get the initial tokenizer I found to work, however :/ For now I'll leave this, cause it seems to work, but I've added a TODO to address this.